lib/p2p/Peer.ts

```diff
@@ -428,7 +412,7 @@ class Peer extends EventEmitter {
       if (this.connected) {
```

Comment:
Assuming we're already connected, this condition can be removed as well.
The problem is that `this.close` won't be called, but this is solved anyway by the connection retries PR.